src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -118,15 +118,29 @@ const TensorIndexMap cast_to_tensor_index_map(const py::dict& inputs) {
 
 namespace string_helpers {
 
+namespace {
+auto find_last_not_null(const std::string& str) {
```

Comment:
could you please specify return type here for readability?

Reply:
It was done intentionally to auto deduce return type. As local scope helper is easier to modify it. The IDE's hint or compiler will show detailed type if there will be wrong use.

Reply:
base() method return type is not immediately clear. So for this code to be readable the function itself should declare the type explicitly. The code must be readable without extra tooling if possible.

Moreover, I suggest to make the code even more readable by replacing all the complex function-style logic with a simple loop that returns a pointer. All these std calls are clearly not necessary for such simple job as finding a zero byte position in an array.